sdks/java/io/hadoop/jdk1.8-tests/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/integration/tests/HIFIOHCatalogIT.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.beam.sdk.io.hadoop.inputformat.custom.options.HIFTestOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.testing.PAssert;
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.io.WritableComparable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hive.hcatalog.data.DefaultHCatRecord;
+import org.apache.hive.hcatalog.mapreduce.HCatInputFormat;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * A test of {@link org.apache.beam.sdk.io.hadoop.inputformat.HadoopInputFormatIO} on an
+ * independent Hive instance using HCatalog.
+ *
+ * <p>This test requires a running instance of Hive, and the test dataset must exist in
+ * the database.
+ *
+ * <p>You can run this test by doing the following:
+ * <pre>
+ *  mvn -e -Pio-it verify -pl sdks/java/io/hadoop/jdk1.8-tests/HIFIOHCatalogIT
+ *  -DintegrationTestPipelineOptions='[
+ *  "--hiveMetastoreUri=thrift://metastoreIp:port"]'
+ * </pre>
+ *
+ * <p>If you want to run this with a runner besides directrunner, there are profiles for dataflow
+ * and spark in the jdk1.8-tests pom. You'll want to activate those in addition to the normal test
+ * runner invocation pipeline options.
+ */
+
+@RunWith(JUnit4.class)
+public class HIFIOHCatalogIT implements Serializable {
+
+  private static final String HIVE_DATABASE = "default";
+  private static final String HIVE_TABLE = "employee";
+  private static final String HIVE_FILTER = "salary=\"1000\"";
+  private static HIFTestOptions options;
+  @Rule
+  public final transient TestPipeline pipeline = TestPipeline.create();
+
+  @BeforeClass
+  public static void setUp() {
+    PipelineOptionsFactory.register(HIFTestOptions.class);
+    options = TestPipeline.testingPipelineOptions().as(HIFTestOptions.class);
+  }
+
+  /**
+   * This test reads data from the Hive instance using HCatalog
+   * and verifies if data is read successfully.
+   * @throws IOException
+   */
+  @Test
+  public void testHIFReadForHCatalog() throws IOException {
+    Long expectedRecordsCount = 200000L;
+    Configuration conf = getConfiguration(options);
+    PCollection<KV<Long, String>> hiveData = pipeline.apply(HadoopInputFormatIO
+        .<Long, String>read().withConfiguration(conf).withValueTranslation(myValueTranslate));
+    PAssert.thatSingleton(hiveData.apply("Count", Count.<KV<Long, String>>globally()))
+        .isEqualTo(expectedRecordsCount);
+    pipeline.run().waitUntilFinish();
+  }
+
+  static SimpleFunction<DefaultHCatRecord, String> myValueTranslate =
```

Comment:
Is it possible to raise the type to HCatRecord here (and in all uses, and in the website PR)?